File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/data/utilities/geometry/PathIterator.java
package com.TritiumGaming.phasmophobiaevidencepicker.data.utilities.geometry;

import java.lang.annotation.Native;

public interface PathIterator {

    @Native int WIND_EVEN_ODD = 0;
    @Native int WIND_NON_ZERO = 1;
    @Native int SEG_MOVETO = 0;
    @Native int SEG_LINETO = 1;
    @Native int SEG_QUADTO = 2;
    @Native int SEG_CUBICTO = 3;
    @Native int SEG_CLOSE = 4;

    int getWindingRule();
    boolean isDone();
    void next();
    int currentSegment(float[] coords);
    int currentSegment(double[] coords);
}

File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/utils/geometry/PathIterator.kt
package com.TritiumGaming.phasmophobiaevidencepicker.utils.geometry

interface PathIterator {
    val windingRule: Int
    val isDone: Boolean
    fun next()
    fun currentSegment(coords: FloatArray): Int
    fun currentSegment(coords: DoubleArray): Int

    companion object {
        const val WIND_EVEN_ODD: Int = 0
        const val WIND_NON_ZERO: Int = 1
        const val SEG_MOVETO: Int = 0
        const val SEG_LINETO: Int = 1
        const val SEG_QUADTO: Int = 2
        const val SEG_CUBICTO: Int = 3
        const val SEG_CLOSE: Int = 4
    }
}
